include/envoy/server/filter_config.h

```diff
@@ -151,6 +154,24 @@ class NamedNetworkFilterConfigFactory {
                                                      FactoryContext& context) PURE;
 
   /**
+   * v2 variant of createFilterFactory(..), where filter configs are specified as proto. This may be
+   * optionally implemented today, but will in the future become compulsory once v1 is deprecated.
+   */
+  virtual NetworkFilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message& config,
+                                                              FactoryContext& context) {
+    UNREFERENCED_PARAMETER(config);
+    UNREFERENCED_PARAMETER(context);
+    return NetworkFilterFactoryCb();
+  }
+
+  /**
+   * @return std::unique_ptr<Protobuf::Message> create empty config proto message for v2. The filter
+   *         config will be parsed into the result of this method. Optional today, will be
+   *         compulsory when v1 is deprecated.
+   */
+  virtual std::unique_ptr<Protobuf::Message> createConfigProto() { return nullptr; }
```

Comment:
Sorry I'm a little confused. I'm trying to understand why we need a createConfigProto() function in addition to the new style factory function. (I think I'm getting mixed up with how we are going from proto to JSON and then back to proto, etc.). I assume this is because we need a well typed structure to jam JSON into, but isn't that somehow known from the type inside it? Is there any way to make this more clear?